### PR TITLE
Add EC scalar mul functions to bssl speed

### PR DESCRIPTION
**Pull request by dkostic**
### Issues:
N/A

### Description of changes: 
Adding lower level EC scalar multiplication operations to bssl speed tool: mul, mul_base, mul_public.

Example output of the new benchmarks:
```
❯ ./tool/bssl speed -filter P-256
Did 21000 ECDH P-256 operations in 1004050us (20915.3 ops/sec)
Did 63000 ECDSA P-256 signing operations in 1009087us (62432.7 ops/sec)
Did 24000 ECDSA P-256 verify operations in 1011729us (23721.8 ops/sec)
Did 28000 ECMUL P-256 mul operations in 1004514us (27874.2 ops/sec)
Did 151000 ECMUL P-256 mul base operations in 1002862us (150569.1 ops/sec)
Did 24000 ECMUL P-256 mul public operations in 1036383us (23157.5 ops/sec)
```

### Call-outs:
Point out areas that need special attention or support during the review process. Discuss architecture or design changes.

### Testing:
How is this change tested (unit tests, fuzz tests, etc.)? Are there any testing steps to be verified by the reviewer?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tool/speed.cc [hunk 1]
```diff
@@ -1080,6 +1080,82 @@ static bool SpeedECDSA(const std::string &selected) {
...
+        return true;
+      })) {
+    return false;
+  }
+  results.Print(name + " mul public");
+
+  return true;
+}
+
+static bool SpeedECMUL(const std::string &selected) {
```
Comment (torben-hansen) [inline tool/speed.cc:1150]: Can we hook up 25519 as well? Not sure API allows it.

File: tool/speed.cc [hunk 2]
```diff
@@ -1790,8 +1866,9 @@ bool Speed(const std::vector<std::string> &args) {
      !SpeedRandom(selected) ||
      !SpeedECDH(selected) ||
      !SpeedECDSA(selected) ||
-     // OpenSSL 1.0 doesn't support Scrypt
 #if !defined(OPENSSL_1_0_BENCHMARK)
```
Comment (torben-hansen) [inline tool/speed.cc:1869]: Some of these curves are surely supported by 1.0.2?

**Review by andrewhop — COMMENTED** [review]
Please include a snippet of speed's text and json output to make sure everything looks good with this change, nothing scientific.

File: tool/speed.cc (hunk 1 above)
Comment (dkostic) [reply tool/speed.cc:1150]: that's already present in `Speed25519` function.

File: tool/speed.cc (hunk 2 above)
Comment (dkostic) [reply tool/speed.cc:1869]: yes, the problem with 1.0.x is it doesn't support `BN_rand_range` and `EC_GROUP_get0_order` functions. That's why I left it out. I don't think the added complexity would be worth it (we don't care about performance of these "low level" functions with 1.0.x).

**Review by torben-hansen — APPROVED** [review]

File: tool/speed.cc (hunk 2 above)
Comment (nebeid) [reply tool/speed.cc:1869]: You may to extend the comment on Scrypt, or in a separate one, to contain what you stated above.

**Review by andrewhop — APPROVED** [review]

**Pull request merged**